gcapi/client.py

```diff
@@ -904,3 +781,249 @@ def add_cases_to_reader_study(
+        descriptions and creates the archive item to be used on the platform.
+
+        Parameters
+        ----------
+        archive
+            slug for the archive (e.g. "i-am-an-archive"). You can find this
+            readily in the URL you use to visit the archive page:
+
+                https://grand-challenge.org/archives/i-am-an-archive/
+
+        archive_items
+            The format for the description of archive items is as follows:
+
+            [
+                {
+                    "slug_0": ["filepath_0", ...],
+                    "slug_1": "filepath_0",
+                    "slug_2": pathlib.Path("filepath_0"),
+                    ...
+                    "slug_n": {"json": "value"}
+
+                },
+                ...
+            ]
+
+            Where the file paths are local paths to the files making up a
+            single image. For file type interfaces the file path can only
+            reference a single file. For json type interfaces any value that
+            is valid for the interface can directly be passed.
+
+        Returns
+        -------
+        The pks of the newly created archive items.
+        """
+
+        if isinstance(archive, str):
+            archive = yield from self.__org_api_meta.archives.detail(
+                slug=archive
+            )
+
+        created_archive_items = yield from self._create_socket_value_sets(
+            creation_kwargs={"archive": archive.api_url},
+            descriptions=archive_items,
+            api=self.__org_api_meta.archive_items,
+        )
+
+        return [ai.pk for ai in created_archive_items]
+
+    def _fetch_socket_detail(
+        self,
+        slug_or_socket,
+        cache=None,
+        ref_url="https://grand-challenge.org/components/interfaces/reader-studies/",
+    ):
+        if isinstance(slug_or_socket, gcapi.models.ComponentInterface):
+            return slug_or_socket
+        else:
+            slug = slug_or_socket
+
+        if cache and slug in cache:
+            return cache[slug]
+        try:
+            interface = yield from self.__org_api_meta.interfaces.detail(
+                slug=slug
+            )
+
+            if cache:
+                cache[slug] = interface
+
+            return interface
+        except ObjectNotFound as e:
+            raise ValueError(
+                f"{slug} is not an existing interface. "
+                f"Please provide one from this list: "
+                f"{ref_url}"
+            ) from e
+
+    def _create_socket_value_sets(
+        self,
+        *,
+        creation_kwargs: dict,
+        descriptions: list[SocketValueSetDescription],
+        api: ModifiableMixin,
+    ):
+        interface_cache: dict[str, gcapi.models.ComponentInterface] = {}
+
+        # Firstly, prepare ALL the strategies
+        strategy_collections: list[list[SocketValueCreateStrategy]] = []
+        for description in descriptions:
+            strategy_per_value = []
```

Comment:
Maybe clearer to call this `strategy_per_value_set`. I was initially confused and thought this referred to CIVs.

Reply:
I've updated a few collection names. Thought I think it IS meant to be a strategy per CIV. Or a strategy that should create CIVs: eventually.

Let me know what you think of the new names.